pydantic/networks.py

```diff
@@ -350,6 +397,98 @@ class PostgresDsn(AnyUrl):
     }
     user_required = True
 
+    __slots__ = AnyUrl.__slots__ + ('hosts',)
+
+    def __init__(self, *args, hosts: Optional[List['HostParts']] = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.hosts = hosts
+
+    @classmethod
+    def validate_host_parts(cls, parts: 'HostParts') -> 'HostParts':
+        """
+        A method used to validate host parts of an URL.
+        """
+        port = parts['port']
+        if port is not None and int(port) > 65_535:
+            raise errors.UrlPortError()
+
+        return parts
+
+    @classmethod
+    def validate_multi_host(cls, hosts: List[str]) -> List['HostParts']:
+        hosts_parts: List['HostParts'] = []
+        for host in hosts:
+            hm = host_regex().match(host)
```

Comment:
with this we're re-running the host regex. Instead of running this bit of the regex twice, surely have just `r'(?P<hosts>.+?)` in `_postgres_url_regex_cache` above, then do the validation here?

Reply:
Makes sense 💪